### ISSUE TITLE
Billing integration

### ISSUE THREAD

**Issue by chicco785** [gravitee-io/issues#1061]
### Expected Behavior

1. When a user subscribe to a plan, he is charged the corresponding costs via a billing solution.

**Comment by NicolasGeraud** [comment]
hi @chicco785 ,
what sort of service do you want/need to integrate ?

**Comment by chicco785** [comment]
Hi,
Haven’t yet done a good scouting. I was wondering if anyone did anything in that direction, before doing anything myself.

**Comment by vinci1it2000** [comment]
I found an interesting [article](https://medium.com/@siftery/top-recurring-billing-and-payments-tools-compared-b2616666f138) that compares `Top Recurring Billing and Payments Tools`. The most used tool for Recurring Billing is [stripe](https://stripe.com/it/billing). It does not have any monthly fee, but it has a fixed fee for all payment methods. The Stripe Billing API is the easiest tool to integrate. It allows  to design fully customized subscription logic and pricing models. The limitation is that, for business, it does not cover all countries, but it accepts payments from anywhere in the world.

**Comment by brasseld** [comment]
Nice shot @vinci1it2000 

I read the article and some others very carefully and it would be a good option for providing first steps of billing integration into Gravitee.

**Comment by vinci1it2000** [comment]
Do you have any news about this issue? Did you implement any billing system to monetize the api?

**Comment by brasseld** [comment]
Hi @vinci1it2000 

No issue for the moment. We do not have time to work on this topic. 
Do you have any suggestion about this one ? Or do you think you can have some time to have a look on the way to integrate stripe or some other billing solution with Gravitee.io ? In that case, I will be really happy to help you !

**Comment by wordtracker** [comment]
There's a big gap in the API management space at the moment. I've done quite a bit of research this past month and 3scale is pretty much the only system out there that integrates billing & rate limiting/quota properly. I'd love to see some other players like Gravitee, Tyk and Kong realise how important the billing piece is and implement it.

**Comment by brasseld** [comment]
Yeah I totally agree. Now, 3scale is not open-source (afaik) and are a plain saas solution, so it's probably more easier to manage billing piece.

An other point when we're talking about billing is about the way to integrate with any payment provider... There is no "specification" we can rely on and we have to choose for the correct provider... which can also take some % on the amounts...

Perhaps that it could be interesting to start with a well-known provider like Stripe to experience the platform and its integration within gravitee...

**Comment by wordtracker** [comment]
Yep, Stripe, exactly. Stripe is the main one that people use. Use that to create an optimum path. Then maybe abstract out the main methods to allow others to plug in different payment providers.

**Comment by dustball** [comment]
FWIW - I found your company form a google search - I have built an API and I wish to monetize it.  I want a system that manages api keys, usage, and billing -- via Stripe. 

The search string was:  "site:github.com monetize api stripe"

See also https://www.quora.com/unanswered/Is-there-a-simple-solution-to-monetize-an-API-with-Stripe-Mashery-and-Apigee-seem-overly-complex-and-enterprisey-I-want-a-simple-solution-to-manage-API-keys-users-and-billing

**Comment by zeluspudding** [comment]
Yeah, this is a huge gap in all of the API management offerings. I would build this into gravitee if only I knew how

**Comment by brasseld** [comment]
The only way to go for now is to tale advantage of analytics report.

API monetization is now part of our roadmap but it would not be available as part of the OSS product.

**Comment by zeluspudding** [comment]
Are you saying to query the analytics report for usage an implement one's own billing atop that?

**Comment by brasseld** [comment]
A sort of, yes

**Comment by gustavovalverde** [comment]
Does this have some kind of ETA (months / years)? Is there a Alpha / Beta / Paid program to test?

**Comment by crisog** [comment]
> The only way to go for now is to tale advantage of analytics report.
> 
> API monetization is now part of our roadmap but it would not be available as part of the OSS product.

Hey, @brasseld any news on this?

**Comment by restyler** [comment]
What a nice thread which I've been tracking for 2 years already.. 
I am bootstrapping ApiRoad.net, which combines a dashboard and reverse proxy (API gateway). As an API vendor on RapidAPI myself, I always wanted to properly implement usage-based API billing and combine it with adequate logs and analytics. It was a challenging project in terms of development efforts and time spent, where I have combined ClickHouse for logs & billing; and Stripe for recurring payments.  I have been using ApiRoad for my own APIs for a few months now, and I am finally starting to roll out vendor access so that anyone can list their API, design plans, and get proper billing for it. I will be happy to chat with people who are seeking a solution for API billing, please drop me a message: contact@apiroad.net

**Comment by mbwmbw1337** [comment]
Would love to see easily billing integration. Any news on this?
